datafusion/core/src/execution/context/mod.rs

```diff
@@ -331,6 +332,15 @@ impl SessionContext {
         self
     }
 
+    /// Adds an analyzer rule to the `SessionState` in the current `SessionContext`.
```

Comment:
Would be nice to have an examples, or doc test for the this method

Reply:
I agree --  we are tracking adding an example for how to use custom analyzer rules in https://github.com/apache/datafusion/issues/10855, so perhaps we can add the example as part of that ticket (i think @goldmedal  said he may have some time to work on that eventually)

Reply:
FWIW I have a WIP PR to improve these examples -- I hope to get it up for review sometime this weekend